controllers/cloud.redhat.com/providers/appinterface_objectstore.go

```diff
@@ -0,0 +1,112 @@
+package providers
+
+import (
+	"fmt"
+	"strings"
+
+	crd "cloud.redhat.com/clowder/v2/apis/cloud.redhat.com/v1alpha1"
+	"cloud.redhat.com/clowder/v2/controllers/cloud.redhat.com/config"
+	"cloud.redhat.com/clowder/v2/controllers/cloud.redhat.com/errors"
+	core "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type AppInterfaceObjectstoreProvider struct {
+	Provider
+	Config config.ObjectStoreConfig
+}
+
+func (a *AppInterfaceObjectstoreProvider) Configure(c *config.AppConfig) {
+	c.ObjectStore = &a.Config
+}
+
+func NewAppInterfaceObjectstore(p *Provider) (ObjectStoreProvider, error) {
+	provider := AppInterfaceObjectstoreProvider{Provider: *p}
+
+	return &provider, nil
+}
+
+func (a *AppInterfaceObjectstoreProvider) CreateBuckets(app *crd.ClowdApp) error {
+	if len(app.Spec.ObjectStore) == 0 {
+		return nil
+	}
+
+	secrets := core.SecretList{}
+	err := a.Client.List(a.Ctx, &secrets, client.InNamespace(app.Namespace))
+
+	if err != nil {
+		msg := fmt.Sprintf("Failed to list secrets in %s", app.Namespace)
+		return errors.Wrap(msg, err)
+	}
+
+	objStoreConfig, err := genObjStoreConfig(secrets.Items)
+
+	if err != nil {
+		return err
+	}
+
+	resolveBucketDeps(app.Spec.ObjectStore, objStoreConfig)
+	a.Config = *objStoreConfig
+	return nil
+}
+
+func resolveBucketDeps(requestedBuckets []string, c *config.ObjectStoreConfig) error {
+	buckets := []config.ObjectStoreBucket{}
+	missing := []string{}
+
+	for _, requestedBucket := range requestedBuckets {
+		found := false
+		for _, bucket := range c.Buckets {
+			if strings.HasPrefix(bucket.Name, requestedBucket) {
+				found = true
+				bucket.RequestedName = requestedBucket
+				buckets = append(buckets, bucket)
+				break
+			}
+		}
+
+		if !found {
+			missing = append(missing, requestedBucket)
+		}
+	}
+
+	if len(missing) > 0 {
+		bucketStr := strings.Join(missing, ", ")
+		return errors.New("Missing buckets from app-interface: " + bucketStr)
+	}
+
+	c.Buckets = buckets
+	return nil
+}
+
+func genObjStoreConfig(secrets []core.Secret) (*config.ObjectStoreConfig, error) {
+	buckets := []config.ObjectStoreBucket{}
+	objectStoreConfig := config.ObjectStoreConfig{Port: 443}
+
+	for _, secret := range secrets {
```

Comment:
This loops through ALL secrets in a namespace looking for AWS related ones. Is there a naming convention we can leverage here? Just feels kinda weird that we are pulling all of them.

Reply:
Can you articulate what specifically feels weird about it?

Reply:
I guess because having those keys doesn't MAKE it an AWS secret, kinda like a Go interface, though it satisfies, doesn't mean it IS of the type we want and I suppose, the endpoint is the most important piece, and we are saying if it doesn't exist we are just going to assume it is one of our secrets?

Reply:
If you're concerned about the endpoint, we can construct that without any secret data as long as we know the region.  We could leverage a naming convention, but I just don't want to force teams to do more than we're already asking.  That said, I should go see what the secret names are just in case they all conform.